src/main/java/frc/robot/RobotContainer.java

```diff
@@ -54,42 +66,106 @@ public RobotContainer() {
 
-        this.driverController.y().onTrue(
+        this.autoChooser = new AutoChooser<>(
+                new AutoOption(
+                        "DoNothing",
+                        autos.doNothing(),
+                        Constants.CompetitionType.COMPETITION
+                )
+        );
+
+        configureAutos();
+        configureButtonBindings();
+    }
+
+    public Command runEjectShooter() {
+        return Commands.deadline(
+                intake
+                        .runStopCommand()
+                        .until(superstructure.atSetpoint)
+                        .withTimeout(6)
+                        .andThen(intake.runEjectInCommand()
+                                .withTimeout(4)),
+                superstructure.toGoal(Superstructure.Goal.EJECT)
+        );
+    }
+
+    public Command runEjectIntake() {
+        return Commands.deadline(
+                intake
+                        .runStopCommand()
+                        .until(superstructure.atSetpoint)
+                        .withTimeout(6)
+                        .andThen(intake.runEjectOutCommand()),
+                superstructure.toGoal(Superstructure.Goal.BACK_FEED)
+        );
+    }
+
+    public Command driveAndAmp() {
+        return Commands.deadline(
                 Commands.sequence(
-                        superstructure.toGoal(Superstructure.Goal.IDLE),
-                        Commands.waitUntil(superstructure.atGoalTrigger),
-                        Commands.waitSeconds(6),
-                        superstructure.toGoal(Superstructure.Goal.SUBWOOFER),
-                        Commands.waitUntil(superstructure.atGoalTrigger),
-                        Commands.waitSeconds(4),
-                        superstructure.toGoal(Superstructure.Goal.IDLE)
+                        Commands.waitUntil(swerve.atHolonomicDrivePose)
+                                .withTimeout(6),
+                        intake.feedHalfCommand(),
+                        Commands.deadline(
+                                Commands.waitUntil(superstructure.atSetpoint)
+                                        .andThen(Commands.waitSeconds(0.5))
+                                        .andThen(intake.feedCommand())
+                                        .andThen(Commands.waitSeconds(0.5)),
+                                superstructure.toGoal(Superstructure.Goal.AMP)
+                        )
+                ),
+                swerve.driveToPose(FieldConstants::getAmpScoringPose)
+        );
+    }
+
+    public Command amp() {
+        return Commands.sequence(
+                intake.feedHalfCommand(),
+                Commands.deadline(
+                        Commands.waitUntil(superstructure.atSetpoint)
+                                .andThen(Commands.waitSeconds(0.5))
+                                .andThen(intake.feedCommand())
+                                .andThen(Commands.waitSeconds(0.5)),
```

Comment:
This time-based stuff is probably fine to leave in for now.

Reply:
Why did we add that wait after it reaches the set point again?